File: octave/interp/lagrange_test.m
% TEST: lagrange interpolation

% 3 points are sufficient to interpolate a polynomial of degree 2
x = [1 2 5];
y = x.^2;
% 100 evenly spaced values between 0 and 100
xx = linspace(0,100);
% interpolate to get yy values
yy = lagrange(xx,x,y);
% pre-computed y values are plotted with 'o',
% interpolated y values are plotted with '.'
plot(x,y,'o',xx,yy,'.')
pause;

File: python/lagrange_test.py
# Lagrange interpolating polynomial test

from lagrange import lagrange
import matplotlib.pyplot as plt
import numpy as np


# Three points are sufficient to interpolate a polynomial of degree 2
x = np.array([15, 42, 30])
y = np.square(x)
X = np.linspace(0, 100, 100)
Y = lagrange(x, y, X)

# Plot the result
plt.scatter(x, y)
plt.plot(X, Y)
plt.show()
